packages/react-components/react-avatar/src/components/Avatar/useAvatar.tsx

```diff
@@ -45,14 +45,17 @@ export const useAvatar_unstable = (props: AvatarProps, ref: React.Ref<HTMLElemen
   );
 
   const [imageHidden, setImageHidden] = React.useState<true | undefined>(undefined);
-  const image: AvatarState['image'] = resolveShorthand(props.image, {
-    defaultProps: {
-      alt: '',
-      role: 'presentation',
-      'aria-hidden': true,
-      hidden: imageHidden,
-    },
-  });
+  // Image shouldn't be resolved if its src is undefined
+  const image: AvatarState['image'] = props.image?.src
+    ? resolveShorthand(props.image, {
+        defaultProps: {
+          alt: '',
+          role: 'presentation',
+          'aria-hidden': true,
+          hidden: imageHidden,
+        },
+      })
+    : undefined;
```

Comment:
Typically you can't access props of a slot shorthand before calling resolveShorthand, in case the `props.image` is a shorthand rather than props object:
```ts
  let image: AvatarState['image'] = resolveShorthand(props.image, {
    defaultProps: {
      alt: '',
      role: 'presentation',
      'aria-hidden': true,
      hidden: imageHidden,
    },
  });

  // Image shouldn't be rendered if its src is not set
  if (!image?.src) {
    image = undefined;
  }  
```

Reply:
I suppose `img` tags don't support shorthand since they can't have children, so actually what you have will probably(?) work fine. However, it might still be cleaner to do it the way I suggested?